securedpreferencestore/src/main/java/devliving/online/securedpreferencestore/SecuredPreferenceStore.java

```diff
@@ -75,16 +73,29 @@ synchronized public static SecuredPreferenceStore getSharedInstance() {
      * @throws InvalidKeyException
      * @throws NoSuchProviderException
      */
-    public static void init( Context appContext,
+    public static void init( Context appContext, String fileName, String keyPrefix,
                              RecoveryHandler recoveryHandler ) throws IOException, CertificateException, NoSuchAlgorithmException, KeyStoreException, UnrecoverableEntryException, InvalidAlgorithmParameterException, NoSuchPaddingException, InvalidKeyException, NoSuchProviderException {
 
         if(mInstance != null){
-            Log.w("SECURED-PREFERENCE", "init called when there already is a non-null instance of the class");
             return;
         }
 
         setRecoveryHandler(recoveryHandler);
-        mInstance = new SecuredPreferenceStore(appContext);
+        mInstance = new SecuredPreferenceStore(appContext, fileName, keyPrefix);
+    }
+
+    /**
+     * @see #init(Context, String, RecoveryHandler)
+     */
+    public static void init( Context appContext, String fileName, RecoveryHandler recoveryHandler) throws IOException, CertificateException, NoSuchAlgorithmException, InvalidKeyException, UnrecoverableEntryException, InvalidAlgorithmParameterException, NoSuchPaddingException, NoSuchProviderException, KeyStoreException {
```

Comment:
for the feature to enable users to change the filename and key prefixes, we need to think about backward compatibility e.g how does these changes affect existing apps that use old versions? how do they upgrade to this version without loosing data?